pkgs/by-name/un/unshieldv3/package.nix

```diff
@@ -0,0 +1,52 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchpatch,
+  cmake,
+  nix-update-script,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "unshieldv3";
+  version = "0.2.1";
+
+  src = fetchFromGitHub {
+    owner = "wfr";
+    repo = "unshieldv3";
+    rev = "v${finalAttrs.version}";
+    sha256 = "sha256-InyWlSJ0u2PqeATb+pd6ELl3x9mhkmD/6gwiUFVc4ZA=";
+  };
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/wfr/unshieldv3/commit/99a5cf5e04e67f37a58620343079b6c6a97e4eb3.patch";
+      hash = "sha256-yElDUxAchMLY784u4i7++YQvUlMGyDfhwCAi6PzExrI=";
+    })
+  ];
+
+  nativeBuildInputs = [ cmake ];
+
+  doInstallCheck = true;
+  installCheckPhase = ''
+    runHook preInstallCheck
+
+    for i in $src/test-data/*.Z; do
+      mkdir -p test
+      $out/bin/unshieldv3 extract $i test
+    done
+
+    runHook postInstallCheck
+  '';
```

Comment:
Maybe that is being pedantic, do you think that this could be done in `checkPhase` or `postCheck`?

Reply:
Sure. I went with `postCheck` to avoid overriding the `checkPhase`, in case the upstream ever adds CMake tests.